docs/source/conf.py

```diff
@@ -13,17 +13,43 @@
 import os
 import sys
 import sphinx_rtd_theme
-import shutil
+from nbconvert import NotebookExporter
+from traitlets.config import Config
 
 ROOT = os.path.abspath(os.path.join("..", ".."))
 sys.path.insert(0, ROOT)
 
-shutil.rmtree(os.path.join(ROOT, "docs/source/read-only-examples"), ignore_errors=True)
-shutil.copytree(
-    os.path.join(ROOT, "examples"),
-    os.path.join(ROOT, "docs/source/read-only-examples"),
-    ignore=shutil.ignore_patterns(".ipynb_checkpoints", "*.txt", "*no-doc*"),
-)
+# Copying and Compiling of Examples
+if not os.path.exists(os.path.join(ROOT, "docs/source/read-only-examples")):
+    os.mkdir(os.path.join(ROOT, "docs/source/read-only-examples"))
+
+# Set up nbconvert configuration to strip empty cells and tables of contents
+c = Config()
+c.NotebookExporter.preprocessors = [
+    "nbconvert.preprocessors.RegexRemovePreprocessor",
+    "nbconvert.preprocessors.ExecutePreprocessor",
+]
+
+c.RegexRemovePreprocessor.enabled = True
+c.RegexRemovePreprocessor.patterns = [".*toc.*", "\\s*\\Z"]
```

Comment:
Could we get a comment explaining the regex used here? I don't know what `\Z` does on the top of my head =).

Also, is there a risk that the first regex strips out content not from a TOC?

Reply:
Fair note, we don't want any potential "au*toc*orrelation" omissions or similar. I've replaced it with `".*Table of Contents.*"` which is a bit more unique. As for the regex, I added some explanations, although I don't know if this is the place to explain lbl what each regex pattern is.

Reply:
Alternatively, can we match on the cell metadata? Removing anything with `"toc": true` as in https://github.com/cosmo-epfl/scikit-cosmo/blob/fd6ed9a41ab058fa2c20f5e2628a2887c2a457ba/examples/PlotGFRE.ipynb#L5-L7

Reply:
According to nbconvert docs, there is not currently a way to preprocess with respect to metadata:
https://nbconvert.readthedocs.io/en/latest/api/preprocessors.html#nbconvert.preprocessors.ExecutePreprocessor

I've been a bit down a rabbit hole with this one, for the record. I think I've tried / looked into:
- passing a filter to nbsphinx (I _could_ write my own file-reading template, but this would be more overkill than the current solution)
- reading cell metadata or autogenerating a tag from the metadata, then filtering based on tags
- making CI yell and shout whenever anyone submits an ill-formatted notebook
- never recompiling the notebooks ever again and formatting them just right

Reply:
There were other attempts but some are a blur

Reply:
ok, fair enough 😄 This should work fine as-is